### DISCUSSION TITLE
Confirm how this library handles 404 from FCM

### DISCUSSION THREAD

**Discussion by marcshilling** [kreait/firebase-php#881] [Q&A]
### Describe the feature you would like to see

Email sent from Firebase today:
> We're writing to inform you that we will be garbage collecting all expired Firebase Cloud Messaging registration tokens and topic subscriptions starting May 15, 2024.
> 
> What you need to know
> Starting May 15, 2024, the [stale](https://notifications.google.com/g/p/ANiao5qrJNSJwWJS9rXnwehJHRTVPN-tJmV50UJubD6z7pkkrdp3DXr4lXL_bBrAp7q-cllIdB7rOoO7P9DYDoC-AZMoF4lXz0ZEMtC0p4eVpM9eie1D8bY240pGAg7IhVm-I8AFOcfIcgCBlrwPueRhhexopjj1YEHkW4XBP0RMOvIPgjQGG7CawHUtdHWnwg8AaJGdfBBK435iUlLQ5TpW7iZM4aTkCQ-gDSy4yeOBC_FC77dvenZZe27S91ju3wVfd2Ol-dz39Zv5UvycwuLfbyRn_P8) tokens and topic subscriptions for devices, which have not been connected to FCM in over 270 days, will be considered as expired and invalid. If a request is sent to an invalid token, it will be rejected with a 404 (UNREGISTERED) error.
> 
> Regarding topic sends or notification campaigns, it is possible that you may observe a decrease in the time taken for fanouts to complete. This could be attributed to the removal of expired subscriptions and a reduction in the number of messages sent.
> 
> This process will help improve fanout performance for your campaigns and identify the undeliverable registration tokens.
> 
> What you need to do
> We recommend you to confirm that your service handles 404 error codes from FCM correctly. Please [delete your record](https://notifications.google.com/g/p/ANiao5oguJogz_z-VU44OUScW2n6s0aoP3f9POFpZ9daz4euknw0yhAp46Sp8wn3dTyJe7frCwKyC2Ctt3tnV879ZUxK8EnT0018H-6AU1x4KEMgJ5FZB46-Fj065VMTWNmzaPykGiGEEkBwWn2tWxe06B8SSHWnlFgJzve7WwL2An3sF9ozf2x0pDGCBriWrwoB3C3JXCDQnbsxNfiBeBYjC_FyGqfzEWorsq5wW6l_xHtof24QhKpMekPRo7luvGbOKSWEtsGbHrH4jpP4utipXXkZNBbZmpiHQ1abu0O1ReboxjQVontRnh_ngepyvvo) of any tokens with a 404 response code because they will not be valid again.

I'm currently catching the `NotFound` errors to detect invalid tokens:
```
try {
    $this->messaging->send($message);
} catch (\Kreait\Firebase\Exception\Messaging\NotFound $e) {
    // Purge these push tokens from the DB
} 
```

Can someone please confirm if this will be impacted by the just announced Firebase change? Or will a change be needed to handle these new 404 errors?

**Comment by jeromegamez** [comment]
I don't know what you mean with "this", but I am confident that the new 404s will be indistinguishable from the old ones 🤞🏻
